centreon/src/Core/HostTemplate/Infrastructure/Repository/DbReadHostTemplateRepository.php

```diff
@@ -193,6 +193,139 @@ public function findByRequestParameter(RequestParametersInterface $requestParame
         return $hostTemplates;
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function findByRequestParametersAndAccessGroups(
+        RequestParametersInterface $requestParameters,
+        array $accessGroups
+    ): array {
+        $this->info('Getting all host templates');
+        if ($accessGroups === []) {
+            return $this->findByRequestParameter($requestParameters);
+        }
```

Comment:
If accessGroups is empty, you must return an empty array, as this list is mandatory.

Reply:
Done